### PR TITLE
attempt to fix CI

### PR DESCRIPTION
**Pull request by djs55**
Using Xcode 8.3 in `circle.yml` should hopefully fix the error:

> carthage: A full installation of Xcode.app 8.3 is required to compile this software.
> Installing just the Command Line Tools is not sufficient.
> Xcode can be installed from the App Store.
> Error: An unsatisfied requirement failed this build.

Removing one of the references to the `version` module in `bridge/github` should fix the error:

> Multiple rules generated for _build/default/bridge/github/version.o

Signed-off-by: David Scott <dave.scott@docker.com>

### CODE REVIEW

**Comment by samoht** [comment]
I have pushed more CI fixes directly to your branch. Let's make this green :-)

File: Makefile [hunk 1]
```diff
@@ -23,8 +23,7 @@ depends:
 	  datakit-ci datakit-bridge-github datakit-bridge-local-git -y
 
 datakit:
-	$(BUILD) -n datakit
-	$(RUNTEST) test/datakit
+	$(BUILD) -p datakit
```
Comment (talex5) [inline Makefile]: Why did this go? Are we testing it elsewhere?

Comment (samoht) [reply Makefile]: We are not testing this. And `jbuilder -n datakit` just says that `-n` is not a valid option.

**Comment by samoht** [comment]
I have fixed the Dockerfiles and updated everything to use the latest version. Not sure why appveyor and CircleCI are complaining, need to have a closer look.

**Comment by samoht** [comment]
I've bumped appveyor and CircleCI to 4.05, hopefully this will fix the last issues.

**Comment by samoht** [comment]
The appveyor built is blocked on:
```
datakit -> hvsock >= 0.8.1 -> duration -> alcotest < 0.8.0
```
which should be solved once https://github.com/ocaml/opam-repository/pull/10770 is updated to the windows repository.

File: Dockerfile [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 #FROM ocaml/opam-dev:alpine-3.5_ocaml-4.04.0
 
 RUN git -C /home/opam/opam-repository fetch origin && \
-    git -C /home/opam/opam-repository reset 482bd5f55 --hard && \
+    git -C /home/opam/opam-repository reset 3bcfe6c8d --hard && \
     opam update -u
 
 ENV OPAMERRLOGLEN=0 OPAMYES=1
-RUN sudo apk add tzdata aspcud gmp-dev perl
+RUN sudo apk update && sudo apk add tzdata aspcud gmp-dev perl
```
Comment (thaJeztah) [inline Dockerfile]: `apk add --no-cache` should work as well?

**Comment by samoht** [comment]
@talex5 would it be possible to remove the CI tests (as they are now tested with the main `Dockerfile.ci`) and to remove the check for the client library on 4.03 (or to use an image with opam2 instead of opam1)?

**Comment by talex5** [comment]
@samoth: I've pushed a fix to https://github.com/talex5/datakit/tree/fix-ci and deployed it.

**Comment by samoht** [comment]
@talex5 Thanks! Now all seems to be green apart the windows builds as we are waiting for https://github.com/fdopen/opam-repository-mingw to be updated.

**Comment by samoht** [comment]
Finally all green! Merging.

**Pull request merged**